src/dataflow/src/repr.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! basically a wrapper around the `datatype` crate
+//! for basic Data Representation
+
+mod relation;
+
+use std::borrow::Borrow;
+use std::slice::SliceIndex;
+
+use api::helper::{pb_value_to_value_ref, value_to_grpc_value};
+use api::v1::Row as ProtoRow;
+use datatypes::data_type::ConcreteDataType;
+use datatypes::types::cast::CastOption;
+use datatypes::types::cast_with_opt;
+use datatypes::value::Value;
+use itertools::Itertools;
+pub(crate) use relation::{RelationDesc, RelationType};
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+
+use crate::expr::error::{CastValueSnafu, EvalError};
+
+/// System-wide Record count difference type.
+pub type Diff = i64;
+
+/// System-wide default timestamp type
+pub type Timestamp = u64;
```

Comment:
I wonder why we don't use `i64` to represent the timestamp, and it represents  the time before the unix epoch.

Reply:
Now changing to use `i64` instead, yes it makes more sense to use i64 than u64